llvm/lib/CodeGen/SelectionDAG/TargetLowering.cpp

```diff
@@ -8755,7 +8755,7 @@ SDValue TargetLowering::expandIS_FPCLASS(EVT ResultVT, SDValue Op,
       ISD::CondCode OrderedOp = IsInverted ? ISD::SETUGE : ISD::SETOLT;
       ISD::CondCode UnorderedOp = IsInverted ? ISD::SETOGE : ISD::SETULT;
 
-      if (isCondCodeLegalOrCustom(IsOrdered ? OrderedOp : UnorderedOp,
+      if (isCondCodeLegalOrCustom(UnorderedOp,
```

Comment:
Considering the todo above, maybe it's worth trying to remove the IsOrdered check and seeing if there's a way to improve the x86 output?

Reply:
I would like to work on this, should I close this PR and raise a new issue?

Also what do you mean by 

> Maybe handle ordered if fabs is free?

in the `TODO`.

Reply:
I mean remove the && !IsOrdered check above, which makes these ternaries not redundant. This is artificially restricted because the results were worse